passboltapi/__init__.py

```diff
@@ -225,32 +242,117 @@ def import_public_keys(self, trustlevel='TRUST_FULLY'):
+                             .data.decode())
                 }
             ],
-            **({"folder_parent_id": folder} if folder else {}),
         }, return_response_object=True)
-        r.raise_for_status()
-        r_json = r.json()
-        new_resource_id = r_json.get("body", {}).get("id")
-        if new_resource_id is None:
-            raise ValueError(f"Unexpected resource creation: {r_json}")
+        r_create.raise_for_status()
+        resource = constructor(PassboltResourceTuple)(r_create.json()["body"])
         if folder:
-            self.move_resource_to_folder(new_resource_id, folder)
-        return r_json
+            # get folder perms
+            if folder.permissions is None:
+                folder = self.read_folder(folder.id)
+            # get users with access to folder
+            lookup_users: Mapping[PassboltUserIdType, PassboltUserTuple] = {
+                user.id: user for user in self.list_users(can_access=folder)
+            }
+            # simulate sharing with folder perms
+            share_payload = {
+                "permissions": [
+                    {
+                        "is_new": True,
+                        "user": lookup_users.get(perm.aro_foreign_key) and lookup_users.get(
+                            perm.aro_foreign_key).username,
+                        **{k: v for k, v in perm._asdict().items() if k != "id"},
+                    } for perm in folder.permissions
+                    if perm.aro == "User" and (
+                            lookup_users.get(perm.aro_foreign_key).gpgkey.fingerprint
+                            != self.user_fingerprint
+                    )
+                ],
+                "secrets": self._encrypt_secrets(password, lookup_users.values())
+            }
+            r_simulate = self.post(f"/share/simulate/resource/{resource.id}.json",
+                                   share_payload, return_response_object=True)
+            r_simulate.raise_for_status()
+
+            r_share = self.put(f"/share/resource/{resource.id}.json", share_payload, return_response_object=True)
+            r_share.raise_for_status()
+
+            self.move_resource_to_folder(resource, folder)
+        return r_create
+
+    def update_resource(self,
+                        resource: PassboltResourceTuple,
+                        name: (str | None) = None,
+                        username: (str | None) = None,
+                        description: (str | None) = None,
+                        uri: (str | None) = None,
+                        resource_type_id: (PassboltResourceTypeIdType | None) = None,
+                        password: Union[None, str] = None):
+        payload = {
+            "name": name,
+            "username": username,
+            "description": description,
+            "uri": uri,
+            "resource_type_id": resource_type_id,
+        }
+        if name is None:
```

Comment:
May be before doing the request we raise an exception if name and password are either None or empty string.